bpf/process/bpf_process_event.h

```diff
@@ -172,6 +172,69 @@ struct cwd_read_data {
 	bool resolved;
 };
 
+static inline __attribute__((always_inline)) long
+cwd_read(struct cwd_read_data *data)
+{
+	struct qstr d_name;
+	struct dentry *parent;
+	struct dentry *vfsmnt_mnt_root;
+	struct vfsmount *root_mnt;
+	struct dentry *root_dentry;
+	const struct path *root = data->root;
+	struct dentry *dentry = data->dentry;
+	struct vfsmount *vfsmnt = data->vfsmnt;
+	struct mount *mnt = data->mnt;
+
+	probe_read(&root_dentry, sizeof(root_dentry), _(&root->dentry));
+	probe_read(&root_mnt, sizeof(root_mnt), _(&root->mnt));
```

Comment:
Do root->dentry and root->mnt change? Do we need to read them on every iteration or can we just keep them at `cwd_read_data` and read them once? 

(I understand that this patch just moves the existing code around, but maybe we can improve it while we are at it?)

Reply:
true, I'll check and add commit on top of the move change